torch/csrc/jit/script/slot.h

```diff
@@ -1,45 +1,51 @@
 #pragma once
 #include <ATen/core/ivalue.h>
+#include <ATen/core/jit_type.h>
+#include <torch/csrc/utils/hash.h>
 
 namespace torch {
 namespace jit {
 namespace script {
 
 // a stable location that can hold an IValue.
-// Currently this is internally implemented as a pointer, but when
-// modules become first-class this will be a pair of  <module_ivalue,
-// slot_number>
+// inside a module.
 struct Slot {
-  friend struct NamedIValue;
-  Slot() : slot_(nullptr) {}
-  Slot(at::IValue* slot) : slot_(slot) {}
+  Slot() {}
```

Comment:
do we need this default ctor?

Reply:
std::unordered_map ends up wanting it for some things, e.g. 
```
foo[bar] = ...;
```
The LHS thing may need to construct a default Slot()